src/command/end2end.rs

```diff
@@ -49,9 +50,18 @@ async fn try_run(cmd: &str, dir: &Utf8Path) -> Result<()> {
         .context(format!("Could not spawn command {cmd:?}"))?;
 
     let mut int = Interrupt::subscribe_any();
-    tokio::select! {
-      _ = int.recv() => {},
-      _ = process.wait() => {}
+    
+    let result : Option<ExitStatus> = tokio::select! {
+      _ = int.recv() => None,
+     result = process.wait() => Some(result?)
+    };
+    if let Some(exit_status) = result {
+        if let Some(exit_code) = exit_status.code() {
+            if exit_code != 0 {
+                return Err(anyhow!("Command terminated with exit code {}", exit_code))
+            }
+        }
     }
     Ok(())
```

Comment:
Your code is perfectly fine, but this can also be written as:
```rust
   tokio::select! {
          _ = int.recv() => Ok(()),
          result = process.wait() => {
            let status = result?;
            if !status.success() {
                bail!("Command terminated with exit code {}", status)
            }
            Ok(())
        }
    }
```
I'm fine with both alternatives, so you chose :)